src/content/en/updates/2017/03/custom-scrollbar.md

```diff
@@ -0,0 +1,380 @@
+about what they do? I certainly didn’t. Scrollbars are an indicator of
+_how much_ of the available content is currently visible and _how much progress_
+you as the reader have made. If you scroll down, so does the scrollbar to
+indicate that you are making progress towards the end. If all the content fits
+into the viewport the scrollbar is either hidden or it is scaled to fill the
+height of the viewport. If the content has 2x the height of the viewport, the
+scrollbar fills ½ of the height of the viewport. Content worth 3x the height of
+the viewport scales the scrollbar to ⅓ of the viewport etc. You see the pattern.
+Instead of scrolling you can also click-and-drag the scrollbar to move through
+the site faster. That’s a surprising amount of behavior for an inconspicuous
+element like that. Let’s fight one battle at a time.
+
+## Step 1: Putting it in reverse
+Okay – we can make elements move slower than the scrolling speed with CSS 3D
+transforms as outlined in the parallax scrolling article. Can we also reverse
+the direction? It turns out we can and that’s our way in for building a
+frame-perfect, custom scrollbar. To understand how this works we need to cover a
+few CSS 3D basics first.
+
+To get any kind of perspective projection in the mathematical sense, you’ll most
+likely end up using
+[homogeneous coordinates](https://en.wikipedia.org/wiki/Homogeneous_coordinates).
+I’m not going into detail what they are and why they work, but you can think of
+them like 3D coordinates with an additional, fourth coordinate called _w_. This
+coordinate should be 1 except if you want to have perspective distortion. So all
+points are 4-dimensional vectors [x, y, z, w=1] and consequently matrices need
+to be 4x4 as well.
+
+One occasion where you can see that CSS uses homogeneous coordinates under the
+hood is when you define your own 4x4 matrices in a transform property using the
+`matrix3d()` function. `matrix3d` takes 16 arguments (because the matrix is
+4x4), specifying one column after the other. So we can use this function to
+manually specify rotations, translations etc. But what it also allows us to do
+is mess around with that _w_ coordinate!
+
+Before we can make use of `matrix3d()`, we need a 3D context – because without a
+3D context there wouldn’t be any perspective distortion and no need for
+homogeneous coordinates. To create a 3D context we need a container with a
+`perspective` and some elements inside that we can transform in the newly
+created 3D space. For
+[example](https://googlechrome.github.io/ui-element-samples/custom-scrollbar/step-1.html):
+
+<img src="/web/updates/images/2017/03/custom-scrollbar/perspective.png"
+  alt="A piece of CSS code that distorts a div using CSS’
+    perspective attribute.">
+
+The elements inside a perspective container are processed as follows:
+
+* Turn each corner (vertex) of an element into homogenous coordinates
+  `[x,y,z,w]` (relative to the perspective container)
```

Comment:
`[x,y,z,w]` (relative to the perspective container)

If I'm understanding this, 'relative...' is important to the operation. Maybe comma instead of parenthesis.